src/protocol/mqtt.rs

```diff
@@ -0,0 +1,129 @@
+//! MQTT Binding Template
+
+use crate::extend::ExtendableThing;
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, skip_serializing_none, OneOrMany};
+
+/// MQTT Control Packet
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash)]
+pub enum ControlPacket {
+    #[serde(rename = "mqv:publish")]
+    Publish,
+    #[serde(rename = "mqv:subscribe")]
+    Subscribe,
+    #[serde(rename = "mqv:unsubscribe")]
+    Unsubscribe,
+}
+
+/// MQTT Quality of service
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash)]
+pub enum QoS {
+    #[serde(rename = "quality:0")]
+    AtMostOnce,
+    #[serde(rename = "quality:1")]
+    AtLeastOnce,
+    #[serde(rename = "quality:2")]
+    ExactlyOnce,
+}
+
+/// Extended fields the Form
+#[serde_as]
+#[skip_serializing_none]
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash, Default)]
+pub struct Form {
+    #[serde(
+        rename = "mqv:retain",
+        skip_serializing_if = "std::ops::Not::not",
+        default
+    )]
+    pub retain: bool,
+    #[serde(rename = "mqv:controlPacket")]
+    pub control_packet: Option<ControlPacket>,
+    #[serde(rename = "mqv:qos")]
+    pub qos: Option<QoS>,
+    #[serde(rename = "mqv:topic")]
+    pub topic: Option<String>,
```

Comment:
Mhh... there's a rough edge for this:
> TopicName is used ony for publication to indicate a single topic, and thus MUST not contain any wildcards.

Reply:
We could open an issue for when we add custom validators in the extensions, but I would not wait to have it to have the protocols.